src/azure-cli/azure/cli/command_modules/appconfig/_client_factory.py

```diff
@@ -10,16 +10,29 @@
 logger = get_logger(__name__)
 
 
+STABLE_API_VERSION = '2022-05-01'
+REPLICATION_API_VERSION = '2022-03-01-preview'
```

Comment:
The guideline for multiapi SDKs is to define a new resource type for appconfiguration in https://github.com/Azure/azure-cli/blob/21d07dc528d283030e5911243c24bef2d2c18d88/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L43

and then specify the API version for different operations in each profile https://github.com/Azure/azure-cli/blob/21d07dc528d283030e5911243c24bef2d2c18d88/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L152

for example, `azure-mgmt-keyvault` uses `2021-04-01-preview` as default API version but uses `2022-07-01` for `vaults_operations`: https://github.com/Azure/azure-cli/blob/21d07dc528d283030e5911243c24bef2d2c18d88/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L180-L182

So appconfiguration's definition could be something like
```
        ResourceType.MGMT_APPCONFIGURATION: SDKProfile('2022-05-01', {
            'replicas': '2022-03-01-preview'
        }),
```

Reply:
Thanks for the detailed explanation. Fixed as suggested.